ikarus/utils/tensorutils.hh

```diff
@@ -251,21 +265,38 @@ auto toVoigt(const Eigen::Matrix<ST, size, size, Options, size, size>& E, bool i
  *
  * The function requires that the size of the Voigt notation vector is valid (1, 3, or 6).
  */
-template <typename ST, int size>
-requires(size == 1 or size == 3 or size == 6)
-auto fromVoigt(const Eigen::Vector<ST, size>& EVoigt, bool isStrain = true) {
-  constexpr int matrixSize = (-1 + ct_sqrt(1 + 8 * size)) / 2;
-  Eigen::Matrix<ST, matrixSize, matrixSize> E;
-  E.diagonal() = EVoigt.template head<matrixSize>();
-
+template <typename ST, int size, int Options, int maxSize>
+requires((size == 1 or size == 3 or size == 6) or
+         ((maxSize == 1 or maxSize == 3 or maxSize == 6) and size == Eigen::Dynamic))
+auto fromVoigt(const Eigen::Matrix<ST, size, 1, Options, maxSize, 1>& EVoigt, bool isStrain = true) {
+  constexpr bool isFixedSized   = (size != Eigen::Dynamic);
   const ST possibleStrainFactor = isStrain ? 0.5 : 1.0;
-  if constexpr (matrixSize == 2)
+
+  const size_t inputSize = isFixedSized ? size : EVoigt.size();
+  const size_t matrixSize =
+      isFixedSized ? (-1 + ct_sqrt(1 + 8 * size)) / 2 : (-1 + static_cast<int>(std::sqrt(1 + 8 * inputSize))) / 2;
+
+  auto E = [&]() {
+    if constexpr (isFixedSized) {
+      Eigen::Matrix<ST, matrixSize, matrixSize> E;
+      E.diagonal() = EVoigt.template head<matrixSize>();
+      return E;
+    } else {
+      Eigen::Matrix<ST, Eigen::Dynamic, Eigen::Dynamic> E;
```

Comment:
This should have the maximum size of 3x3 and not a dynamically allocated object.